lib/src/test/java/io/ably/lib/util/CryptoTest.java

```diff
@@ -63,4 +69,108 @@ public void cipher_params() throws AblyException, NoSuchAlgorithmException {
+	 *
+ 	 * The key, IV and message data are the same for every test run so that the
+ 	 * encrypted data may be exported from the console output for consumption by tests
+ 	 * run on other platforms.
+ 	 *
+	 * Equivalent to the following in ably-cocoa:
+	 * testEncryptAndDecrypt in Spec/CryptoTest.m
+	 * @throws IOException 
+	 */
+	@Test
+	public void encryptAndDecrypt() throws NoSuchAlgorithmException, AblyException, IOException {
+		final byte[] key = {
+			1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+			11, 12, 13, 14, 15, 16, 17, 18, 19, 20,
+			21, 22, 23, 24, 25, 26, 27, 28, 29, 30,
+			31, 32,
+		};
+		final byte[] iv = {
+			16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1,
+		};
+		final CipherParams params = Crypto.getDefaultParams(key, iv);
+
+		// Prepare message data.
+		final int maxLength = 70;
+		final byte[] message = new byte[maxLength];
+		for (byte value=1; value<=maxLength; value++) {
+			message[value - 1] = value;
+		}
+
+		final StringWriter target = new StringWriter();
+		final JsonWriter writer = new JsonWriter(target);
+		writer.setIndent("  ");
+		writer.beginObject();
+
+		writer.name("algorithm");
+		writer.value("aes");
+
+		writer.name("mode");
+		writer.value("cbc");
+		
+		writer.name("keyLength");
+		writer.value(256);
+		
+		writer.name("key");
+		writer.value(Base64Coder.encodeToString(key));
+		
+		writer.name("iv");
+		writer.value(Base64Coder.encodeToString(iv));
+
+		// Perform encrypt and decrypt on message data trimmed at all lengths up
+		// to and including maxLength.
+		writer.name("items");
+		writer.beginArray();
+		for (int i=1; i<=maxLength; i++) {
+			// We need to create a new ChannelCipher for each message we encode,
+			// so that our IV gets used (being start of CBC chain).
+			final ChannelCipher cipher = Crypto.getCipher(new ChannelOptions() {{ encrypted=true; cipherParams=params; }});
+
+			// Encrypt i bytes from the start of the message data.
+			final byte[] encoded = Arrays.copyOfRange(message, 0, i);
+			final byte[] encrypted = cipher.encrypt(encoded);
+
+			// Add encryption result to results in format ready for fixture.
+			writeResult(writer, encoded, encrypted);
+
+			// Decrypt the encrypted data and verify the result is the same as what
+			// we submitted for encryption.
+			final byte[] verify = cipher.decrypt(encrypted);
+			assertArrayEquals(verify, encoded);
+		}
+		writer.endArray();
+
+		writer.endObject();
+
+		System.out.println("Fixture JSON for test-resources:\n" + target.toString());
+	}
+
+	private void writeResult(final JsonWriter writer, final byte[] encoded, final byte[] encrypted) throws IOException {
+		writer.beginObject();
+
+		writer.name("encoded");
+		writeData(writer, encoded, null);
+
+		writer.name("encrypted");
+		writeData(writer, encoded, "cipher+aes-256-cbc");
+
+		writer.endObject();
+	}
+
+	private static final String BASE64 = "base64";
+	private void writeData(final JsonWriter writer, final byte[] data, final String encoding) throws IOException {
```

Comment:
Doesn't there need to be something in here that writes the msgpack-formatted message as well?

Reply:
Nothing that I've touched yet that has to read these has needed that property.

Reply:
Nothing that I've touched, so far, has needed that property.